crates/solvers/src/domain/order.rs

```diff
@@ -73,3 +93,87 @@ impl<'a> NonLiquidity<'a> {
         self.0
     }
 }
+
+/// A COW reward amount, in base units.
+#[derive(Clone, Copy, Debug)]
+pub struct Reward(pub f64);
+
+#[derive(Debug, Clone)]
+pub struct CustomInteraction {
```

Comment:
Imagine being new to the code and stumbling upon a `CustomInteraction` or a `JitOrder`. You'd have _no clue_ what these are. I think doc comments on these would be super-valuable.